### DISCUSSION TITLE
Why sometimes "too many requests" error is raised?

### DISCUSSION THREAD

**Discussion by copyNdpaste** [snapshot-labs/snapshot#2213] [Q&A]
I checked snapshot proposals every minute by using aws lambda and the "too many requests" is raised so I changed 1 minute to 3 minutes but error is still not disappeared.

How to solve it?

![image](https://user-images.githubusercontent.com/18900976/164141364-41fd5abb-4a75-4778-900a-e45343c726f3.png)

**Comment by ChaituVR** [comment]
Queries are capped per IP, 64 requests max in 16 seconds
https://github.com/snapshot-labs/snapshot-hub/blob/master/src/helpers/rateLimit.ts

**Reply by copyNdpaste** [reply]
But I think I am not sending more than 64 requests in 16 seconds.
My lambda function is scheduled on each 3 minutes.

**Reply by ChaituVR** [reply]
The rate limit is based on the IP address, maybe some other lambda function got the same IP address and caused this rate limit? Is there a way to generate a new IP address on AWS, not 100% sure but maybe something like the below could help
https://docs.aws.amazon.com/prescriptive-guidance/latest/patterns/generate-a-static-outbound-ip-address-using-a-lambda-function-amazon-vpc-and-a-serverless-architecture.html
